File: src/main/java/dev/paigewatson/layoutmaster/client/services/MongoFreightCarService.java
package dev.paigewatson.layoutmaster.client.services;

import dev.paigewatson.layoutmaster.data.RollingStockDAL;
import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation;
import dev.paigewatson.layoutmaster.models.rollingstock.RollingStock;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class MongoFreightCarService implements FreightCarService
{
    private final RollingStockDAL rollingStockDAL;


    public MongoFreightCarService(@Autowired RollingStockDAL rollingStockDAL)
    {
        this.rollingStockDAL = rollingStockDAL;
    }

    @Override
    public List<RollingStock> allFreightCars()
    {
        return rollingStockDAL.getAllRollingStock();
    }

    @Override
    public List<RollingStock> allFreightCarsByAARType(AARDesignation aarDesignation)
    {
        return rollingStockDAL.getAllOfAARDesignation(aarDesignation);
    }

    @Override
    public void delete(RollingStock rollingStockToDelete)
    {
        rollingStockDAL.delete(rollingStockToDelete);
    }

    @Override
    public List<RollingStock> allFreightCarsByRoadName(String roadName)
    {
        return rollingStockDAL.getAllOfRoadName(roadName);
    }

    @Override
    public RollingStock saveFreightCarToDatabase(RollingStock freightCarToSave)
    {
        return rollingStockDAL.insertRollingStock(freightCarToSave);
    }
}

File: src/main/java/dev/paigewatson/layoutmaster/client/services/MongoFreightCarService.kt
package dev.paigewatson.layoutmaster.client.services

import dev.paigewatson.layoutmaster.data.RollingStockDAL
import dev.paigewatson.layoutmaster.models.rollingstock.AARDesignation
import dev.paigewatson.layoutmaster.models.rollingstock.RollingStock
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class MongoFreightCarService(@param:Autowired private val rollingStockDAL: RollingStockDAL) : FreightCarService {
    override fun allFreightCars(): List<RollingStock> {
        return rollingStockDAL.allRollingStock
    }

    override fun allFreightCarsByAARType(aarDesignation: AARDesignation): List<RollingStock> {
        return rollingStockDAL.getAllOfAARDesignation(aarDesignation)
    }

    override fun delete(rollingStockToDelete: RollingStock) {
        rollingStockDAL.delete(rollingStockToDelete)
    }

    override fun allFreightCarsByRoadName(roadName: String): List<RollingStock> {
        return rollingStockDAL.getAllOfRoadName(roadName)
    }

    override fun saveFreightCarToDatabase(freightCarToSave: RollingStock): RollingStock {
        return rollingStockDAL.insertRollingStock(freightCarToSave)
    }
}
